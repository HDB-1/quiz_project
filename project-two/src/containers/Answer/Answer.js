import React, { Component } from "react";

export default function Answer (props) {
  
    return (
      <div>
        <input id={props.id} type="button" value={atob(props.answer)} name="answer" onClick={() => props.submit(props.answer)}/>
      </div>
    )
  }
