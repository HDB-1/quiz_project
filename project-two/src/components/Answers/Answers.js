import React from "react";
import Answer from "../../containers/Answer/Answer";

export default function Answers(props) {
  let answers = [props.correct_answer, ...props.incorrect_answers];

  for (let i = answers.length - 1; i >= 0; i--) {
    let arrayItemForSwitching = answers[i];
    let targetIndex = Math.floor(Math.random() * (answers.length - 1));
    answers[i] = answers[targetIndex];
    answers[targetIndex] = arrayItemForSwitching;
  }

  return (
    <div>
      {answers.map((answer, index) => {
        return <Answer key={index} id={index} answer={answer} submit={props.submit} />;
      })}
    </div>
  );
}
