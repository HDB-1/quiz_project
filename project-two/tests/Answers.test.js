import React from 'react';
import { shallow } from 'enzyme';
import Answers from '../src/components/Answers/Answers'
import Answer from '../src/containers/Answer/Answer'

describe('Shallow Answers', () => {
  let wrapper;

    const answers = ["1", "2", "3", "4"]

  beforeEach(() => wrapper = shallow(<Answers answers={answers}/>)); // this automatically tests for rendering without crashing.
  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should render 4 answer components', () => {
    expect(wrapper.find(Answer).length).toEqual(4);
  });
});