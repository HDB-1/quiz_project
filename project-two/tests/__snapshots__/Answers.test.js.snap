// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Answers Should render correctly 1`] = `
<div>
  <Answer
    answer="John Quincy Adams"
    key="0"
  />
  <Answer
    answer="Thomas Jefferson"
    key="1"
  />
  <Answer
    answer="Martin Van Buren"
    key="2"
  />
  <Answer
    answer="Ulysses Grant"
    key="3"
  />
</div>
`;
